src/bindings/system/fs.rs

```diff
@@ -190,6 +193,26 @@ pub fn init(lua: &Lua) -> ::Result<()> {
             .map_err(LuaError::external)
     })?)?;
 
+	module.set("copy_file", lua.create_function(|_, (src, dest): (String, String)| {
+		let mut dest = dest;
+		if Path::new(&dest).is_dir() {
+			let file_name = match Path::new(&src)
+				.file_name() {
+					Some(s) => s.to_string_lossy().to_string(),
+					None => return Err(io::Error::from(io::ErrorKind::NotFound))
+										.map_err(LuaError::external)
+			};
+			dest += file_name.as_str();
```

Comment:
The problem with this check is that it is only appending the filename to `dest` so if the directory is called `log` and the file is called `info.log` which would make it `dest` be `loginfo.log` instead of `log/info.log`. The best thing to do is to use either `Path::join` or `PathBuf::push`. Here is a function you can use

```
fn copy_file<S: AsRef<Path>, D: AsRef<Path>>(src: S, dest: D) -> LuaResult<()> {
    let mut dest = dest.as_ref().to_path_buf();
    if dest.is_dir() {
        let file_name = src.as_ref()
            .file_name()
            .map(|s| s.to_string_lossy().to_string())
            .ok_or(LuaError::external(io::Error::from(io::ErrorKind::NotFound)))?; //This would need to be replace with a proper error than using `io::Error`
        dest.push(file_name);
    }
    fs::copy(src, dest).map(|_| ())
        .map_err(LuaError::external)
}
```

Then what you could do from there is do the following
```
module.set("copy_file", lua.create_function(|_, (src, dest): (String, String)| {
        copy_file(src, dest)
})?)?;
```